oauth2-server-core/src/main/java/nl/myndocs/oauth2/CallRouter.kt

```diff
@@ -42,76 +43,36 @@ class CallRouter(
         }
 
         try {
-            val allowedGrantTypes = setOf(PASSWORD, AUTHORIZATION_CODE, REFRESH_TOKEN, CLIENT_CREDENTIALS)
             val grantType = callContext.formParameters["grant_type"]
                     ?: throw InvalidRequestException("'grant_type' not given")
 
+            val grantingCall = object: GrantingCall {
+                override val callContext: CallContext
+                    get() = callContext
+
+                override val tokenService = this@CallRouter.tokenService
+            }
+
+            val granterMap = mutableMapOf<String, Granter>()
```

Comment:
Just as a tiny detail this could probably be done like so, to avoid mutability and all that :)

```kt
val granterMap = granters
    .map {
        val granter = grantingCall.it()
        granter.grantType to granter
    }
    .toMap()
```

Reply:
Good catch. Solved